Assets/Scripts/Town/Interactable.cs

```diff
@@ -0,0 +1,9 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public interface Interactable
+{
+    void OnCanBeInteractedWithBegin();
+    void OnCanBeInteractedWithEnd();
```

Comment:
These method names does not tell us what is going to happen. `On` suggests that they will handle some event, but these functions will be called during normal execution. Simple name like `BeginInteraction` or `EndInteraction` would be more descriptive and shorter.
What is the use case for begin and end? Don't we only need BeginInteraction?

Reply:
The idea was that these methods would be called when the player can interact with an item. I haven't implemented the interaction part of the interface, because I just wanted to test the highlighting. The begin method is called when the player begins to be able to interact with an item (for the trash can the begin method highlights it) and when the player gets out of reach of the object then the end method is called (for the trash can this disables the highlighting).

Reply:
What would be a better way to call these methods?

Reply:
In this case they are ok, I would also think about something like `EnteredInteractionRange`, `LeftInteractionRange`, just to remove this `on` part.